compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -121,26 +121,72 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
     }
 
     def genTopLevelExports(classSym: Symbol): List[js.TopLevelExportDef] = {
-      val allTopLevelExports = for {
+      for (m <- genTopLevelOrStaticExports(classSym, topLevels = true))
+        yield js.TopLevelExportDef(m)(m.pos)
+    }
+
+    def genStaticExports(classSym: Symbol): List[js.Tree] =
+      genTopLevelOrStaticExports(classSym, topLevels = false)
+
+    private def genTopLevelOrStaticExports(classSym: Symbol,
+        topLevels: Boolean): List[js.Tree] = {
+      val exportsNamesAndPositions = {
+        genTopLevelOrStaticFieldExports(classSym, topLevels) ++
+        genTopLevelOrStaticMethodExports(classSym, topLevels)
+      }
+
+      for {
+        (head :: tail) <- exportsNamesAndPositions.tails
+        duplicate <- tail if duplicate._2 == head._2
+      } {
+        reporter.error(duplicate._3,
+            s"Duplicate ${if (topLevels) "top-level" else "static"} export " +
+            s"with name '${duplicate._2}'")
+      }
+
+      exportsNamesAndPositions.map(_._1)
+    }
+
+    private def genTopLevelOrStaticFieldExports(classSym: Symbol,
+        topLevels: Boolean): List[(js.FieldDef, String, Position)] = {
+      (for {
+        fieldSym <- classSym.info.members
+        if !fieldSym.isMethod && fieldSym.isTerm && !fieldSym.isModule
+        export <- jsInterop.registeredExportsOf(fieldSym)
+        if export.isTopLevel == topLevels
+      } yield {
+        assert(export.isStatic) // in fact, top-level field exports do not exist yet
+
+        implicit val pos = fieldSym.pos
+
+        val mutable = true // static fields must always be mutable
+        val name = js.StringLiteral(export.jsName)
+        val irTpe = genExposedFieldIRType(fieldSym)
+
+        (js.FieldDef(static = true, name, irTpe, mutable), export.jsName, pos)
```

Comment:
I'm not sure about this one. This place doesn't seem to be the right place to introduce this comment (after all, for instance fields we also do not mention that). But also I did not find an appropriate place.